raiden/tests/integration/rpc/assumptions/test_rpc_gas_price_assumptions.py

```diff
@@ -50,9 +54,125 @@ def test_duplicated_transaction_same_gas_price_raises(deploy_client):
+    """
+    normal_gas_proxy, _ = deploy_rpc_test_contract(deploy_client, "RpcTest")
+
+    gas_price = GasPrice(0)
+    gas_price_strategy = make_fixed_gas_price_strategy(gas_price)
+    deploy_client.web3.eth.setGasPriceStrategy(gas_price_strategy)
+    zero_gas_proxy = deploy_client.new_contract_proxy(
+        abi=normal_gas_proxy.contract.abi, contract_address=normal_gas_proxy.contract_address
+    )
+
+    address = normal_gas_proxy.contract_address
+    assert len(deploy_client.web3.eth.getCode(to_checksum_address(address))) > 0
+
+    check_block = deploy_client.get_checking_block()
+    gas_estimate = zero_gas_proxy.estimate_gas(check_block, "ret")
+    assert gas_estimate, "Gas estimation should not fail here"
+
+    zerogas_txhash = zero_gas_proxy.transact("ret", gas_estimate)
+
+    zerogas_tx = deploy_client.web3.eth.getTransaction(zerogas_txhash)
+
+    # wait for a few blocks, since this is a private chain, if the transaction
+    # is supposed to be mined it will happen quite fast.
+    current_block = deploy_client.block_number()
+    target_block_number = current_block + 5
+    while current_block < target_block_number:
+        current_block = deploy_client.block_number()
+        gevent.sleep(0.5)
+
+    zerogas_tx = deploy_client.web3.eth.getTransaction(zerogas_txhash)
+
+    msg = "Even thought the transaction had a zero gas price, it is not removed from the pool"
+    assert zerogas_tx is not None, msg
+
+    zerogas_receipt = deploy_client.web3.eth.getTransactionReceipt(zerogas_txhash)
+
+    msg = "Even though the transaction had gas price of zero, it did get mined."
+    assert zerogas_receipt["status"] != RECEIPT_FAILURE_CODE, msg
+
+
+@pytest.mark.parametrize("blockchain_number_of_nodes", [2])
+def test_remote_transaction_with_zero_gasprice_is_not_mined(
+    web3: Web3, deploy_key: PrivateKey, blockchain_rpc_ports: List[Port], blockchain_type: str
+) -> None:
+    """ If the non-local transaction is sent with a gas price set to zero it is
+    not mined.
+    """
+    host = "127.0.0.1"
+    miner_rpc_port, rpc_port = blockchain_rpc_ports
+
+    miner_web3 = Web3(HTTPProvider(f"http://{host}:{miner_rpc_port}"))
+    miner_client = JSONRPCClient(miner_web3, deploy_key)
+
+    web3 = Web3(HTTPProvider(f"http://{host}:{rpc_port}"))
+    client = JSONRPCClient(web3, deploy_key)
+
+    before_deploy_block = client.block_number()
+    normal_gas_proxy, _ = deploy_rpc_test_contract(client, "RpcTest")
+    num_blocks_to_wait = client.block_number() - before_deploy_block
+
+    gas_price = GasPrice(0)
+    gas_price_strategy = make_fixed_gas_price_strategy(gas_price)
+    client.web3.eth.setGasPriceStrategy(gas_price_strategy)
+    zero_gas_proxy = client.new_contract_proxy(
+        abi=normal_gas_proxy.contract.abi, contract_address=normal_gas_proxy.contract_address
+    )
+
+    # assert miner_client.web3.eth.mining, "Miner should be mining"
+    # assert not client.web3.eth.mining, "Extra client should NOT be mining"
```

Comment:
I did not figure out why the non-mining node switched state after the contract is deployed.